services/navigation/builtin/builtin.go

```diff
@@ -575,7 +581,75 @@ func (svc *builtIn) waypointIsDeleted() bool {
 func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{}) ([]*spatialmath.GeoObstacle, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return svc.obstacles, nil
+
+	// get static geoObstacles
+	geoObstacles := svc.obstacles
+
+	for _, detector := range svc.motionCfg.ObstacleDetectors {
+		// get the vision service
+		visSvc, ok := svc.visionServicesByName[detector.VisionServiceName]
+		if !ok {
+			return nil, fmt.Errorf("vision service with name: %s not found", detector.VisionServiceName)
+		}
+
+		// get the detections
+		detections, err := visSvc.GetObjectPointClouds(ctx, detector.CameraName.Name, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// get current geo position of robot
+		gp, _, err := svc.movementSensor.Position(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// instantiate a localizer and use it to get our current position
+		localizer := motion.NewMovementSensorLocalizer(svc.movementSensor, gp, spatialmath.NewZeroPose())
+		currentPIF, err := localizer.CurrentPosition(ctx)
+		if err != nil {
+			return nil, err
+		}
+		// convert orientation of currentPIF to be left handed
+		heading := math.Mod(math.Abs(currentPIF.Pose().Orientation().OrientationVectorDegrees().Theta-360), 360)
+
+		// convert geo position into GeoPose
+		relativeToGeoPose := spatialmath.NewGeoPose(gp, heading)
+
+		// iterate through all detections and construct a geoObstacle to append
+		for i, detection := range detections {
+			// get the geometry's lat & lng along with its heading with respect to north as a left handed value
+			geomGeoPose := spatialmath.PoseToGeoPose(relativeToGeoPose, detection.Geometry.Pose())
+
+			// prefix the label of the geometry so we know it is transient and add extra info
+			label := "transient_" + strconv.Itoa(i) + "_" + detector.CameraName.Name
+			if detection.Geometry.Label() != "" {
+				label += "_" + detection.Geometry.Label()
+			}
+			detection.Geometry.SetLabel(label)
+
+			// determine the desired geometry pose
+			desiredPose := spatialmath.NewPose(
```

Comment:
I don't understand why this is is the desired pose. It is an empty r3.Vector with an orientation facing the same direction as the detected geometry right?

Why do we need to go from pose -> geopose -> back to pose? Why not reuse the orientation of the detected point cloud?

Reply:
> Why not reuse the orientation of the detected point cloud?

The orientation of the detection is from the point of view of the observer. This does not account for any rotation the observer may have.

> Why do we need to go from pose -> geopose -> back to pose? 

we use pose + current geo position to get the geo pose of the detection.